src/c/c_dataset.cpp

```diff
@@ -30,163 +30,196 @@
 
 using namespace SmartRedis;
 
+// Create a new DataSet.
+// The user is responsible for deallocating the DataSet via DeallocateeDataSet()
 extern "C"
-void* CDataSet(const char* name, const size_t name_length)
+void *CDataSet(const char *name, const size_t name_length)
 {
-  /* Return a pointer to a new DataSet.
-  The user is responsible for deleting the client.
-  */
-  std::string name_str = std::string(name, name_length);
-  DataSet* dataset = new DataSet(name_str);
-  return (void*)dataset;
+  // Sanity check parameters
+  if (NULL == name)
+    return NULL;
+
+  std::string name_str(name, name_length);
+
+  DataSet *dataset = NULL;
```

Comment:
I know in pure c programs the convention is usually that the ``*`` is attached to the variable name, but the convention in SmartRedis has been to follow C++ norms (ref: https://www.stroustrup.com/bs_faq2.html#whitespace)  and to attach the ``*`` to the type.  I can see us potentially wanting to have the c-style function prototypes reflect c-standards, but I think internal code (especially code in ``.cpp`` files), should follow C++ conventions.

Reply:
I was thinking that the C interface should follow C style and the C++ interface could follow the new C++ style, but I'm open to discussion.

Reply:
Yeah, I understand that reasoning.  I think given that this is C++ code though (uses standard containers and all) we should follow C++ standards.  Also, helps to have few conventions in the code.

Reply:
I can see that. Consider it done.